### DISCUSSION TITLE
Battery Power vs. consumption

### DISCUSSION THREAD

**Discussion by MikeTango** [WillCodeForCats/solaredge-modbus-multi#579] [General]
**Describe the current behavior**
Hi,

first of all thanks for the awesome work. I rarely have any linux stuff that just werks. What i am seeing is that the templated sensors disagree quite a bit on some values, i guess due to the behaviour of the "native" sensors in the system. At night, under battery power, the calculated sensor.power_battery_discharging reads much higher than the calculated sensor.power_consumption. I don't know if this can be attributed to inverter loss.

**What should be updated or changed?**
Possibly making a template sensor that fails over to the higher value at bat discharge. If it is inverter loss, this is moot.

**Home Assistant (please complete the following information):**
 - Home Assistant Core Version: newest
 - solaredge-modbus-multi Version: newest

**Additional context**
Add any other context about the request here.

**Comment by Ashpork** [comment]
There will be some inverter loss, but it'll be a small % of the power.

It's also worth identifying that whilst the read occurs using this integration, the inverter will propagate it out to the followers, meaning the values from each won't be at the exact same moment. So if you have a very variable load in the house, you will get some additional disparity between your b1 power and your local load.

What templates are you using for the power batter discharging and power consumption? Does the value there look different to b1 power (assuming you have just a single battery?)

To put some context behind it, how much difference are you finding?

**Comment by MikeTango** [comment]
Hi,

i am very new to this and the solar system isn't even two days old, so there's not too much data. Here's last night. The rhythmic highs (and the crazy transients) are probably from the fridge(s) or the hot water circulation pump. Battery is always ~100W above the consumption.

![Screenshot 2024-04-08 at 08-47-00 Verlauf – Home Assistant](https://github.com/WillCodeForCats/solaredge-modbus-multi/assets/3368005/db8e7b7b-4666-40cb-9ca9-ba54d8d68408)
![Screenshot 2024-04-08 at 08-46-29 Verlauf – Home Assistant](https://github.com/WillCodeForCats/solaredge-modbus-multi/assets/3368005/52093474-21f1-46f5-a25f-95d66ea6e8b6)

**Comment by jongbj1972** [comment]
I have a similar issue where the battery export sensor only records approx 3kWh of discharge overnight while in practice it is approx. 15kWh. The latter is correct as that is a full battery charge and the integration shows that this amount of power is added to the battery during the day.

**Comment by MikeTango** [comment]
That sounds like you have a stack of three batteries but are only using the data of one.

**Comment by WillCodeForCats** [comment]
This isn't an integration issue so I'm going to move it to discussions.

Templates in the wiki are contributed (not by me): please contribute new templates as appropriate. I don't have batteries personally so I can't help.
